stream.go

```diff
@@ -0,0 +1,120 @@
+package dht
+
+import (
+	"context"
+	"sync"
+
+	pbio "github.com/gogo/protobuf/io"
+	pb "github.com/libp2p/go-libp2p-kad-dht/pb"
+	inet "github.com/libp2p/go-libp2p-net"
+	peer "github.com/libp2p/go-libp2p-peer"
+	"golang.org/x/xerrors"
```

Comment:
what does this package offer over plain errors/fmt?

Reply:
`xerrors` is the future of Go error handling, see https://go.googlesource.com/proposal/+/master/design/29934-error-values.md for example. There's a lot more literature since on it.